nix/ogmios-nixos-module.nix

```diff
@@ -0,0 +1,120 @@
+# NixOS module for configuring Ogmios service.
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.ogmios;
+in
+with lib; {
+  options.services.ogmios = with types; {
+    enable = mkEnableOption "Ogmios lightweight bridge interface for cardano-node";
+
+    package = mkOption {
+      description = "Ogmios package";
+      type = package;
+      default = pkgs.ogmios;
+    };
+
+    user = mkOption {
+      description = "User to run Ogmios service as.";
+      type = str;
+      default = "ogmios";
+    };
+
+    group = mkOption {
+      description = "Group to run Ogmios service as.";
+      type = str;
+      default = "ogmios";
+    };
+
+    nodeSocket = mkOption {
+      description = "Path to cardano-node IPC socket.";
+      type = str;
+      default = "/run/cardano-node/node.socket";
+    };
+
+    nodeConfig = mkOption {
+      description = "Path to cardano-node config.json file.";
+      type = str;
+    };
+
+    host = mkOption {
+      description = "Host address or name to listen on.";
+      type = str;
+      default = "localhost";
+    };
+
+    port = mkOption {
+      description = "TCP port to listen on.";
+      type = port;
+      default = 1337;
+    };
+
+    extraArgs = mkOption {
+      description = "Extra arguments to ogmios command.";
+      type = listOf str;
+      default = [ ];
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = mkIf (config ? services.cardano-node && config.services.cardano-node.enable) [{
+      assertion = config.services.cardano-node.systemdSocketActivation;
+      message = "The option services.cardano-node.systemdSocketActivation needs to be enabled to use Ogmios with the cardano-node configured by that module. Otherwise cardano-node socket has wrong permissions.";
+    }];
```

Comment:
I think you can rewrite this without `mkIf` using the logical implication `->`. Not sure it would be more readable though.